core/string/ustring.cpp

```diff
@@ -4280,12 +4274,12 @@ String String::pad_decimals(int p_digits) const {
 		c = s.length() - 1;
 	} else {
 		if (p_digits <= 0) {
-			return s.substr(0, c);
+			return s.left(c);
 		}
 	}
 
 	if (s.length() - (c + 1) > p_digits) {
-		return s.substr(0, c + p_digits + 1);
+		return s.left(c + p_digits + 1);
```

Comment:
`c` is the result of `find()`, however, if it's `-1`, it goes into conditions that force it to be at least 0. On the other hand, if `p_digits` isn't greater than 0, it returns early. So this will always be positive, and it's safe.